storage/innobase/ut/ut0crc32.cc

```diff
@@ -342,11 +343,21 @@ allocations, would not hurt if called twice, but would be pointless. */
 void ut_crc32_init()
 {
 #ifndef HAVE_CRC32_VPMSUM
-# if defined(__GNUC__) && defined(HAVE_ARMV8_CRC) && defined(HAVE_ARMV8_CRYPTO)
+# if defined(__GNUC__) && defined(HAVE_ARMV8_CRC)
   if (crc32c_aarch64_available())
   {
     ut_crc32_low= crc32c_aarch64;
-    ut_crc32_implementation= "Using ARMv8 crc32 + pmull instructions";
+#  if defined(HAVE_ARMV8_CRYPTO)
+    pmull_aarch64_runtime_flag = pmull_aarch64_available();
+#  endif
+    if (pmull_aarch64_runtime_flag)
+    {
+      ut_crc32_implementation= "Using ARMv8 crc32 + pmull instructions";
+    }
+    else
+    {
+      ut_crc32_implementation= "Using ARMv8 crc32 instructions";
+    }
     return;
   }
```

Comment:
I think that it would be cleaner to return a `const char*` from the function, identifying the implementation, and `NULL` when no special implementation is availble:
```c++
# if defined(__GNUC__) && defined(HAVE_ARVM8_CRC)
  if (const char *crc32c_implementation= crc32_aarch64_available())
  {
    ut_crc32_low= crc32c_aarch64;
    ut_crc32_implementation= crc32c_implementation;
    return;
  }
```